### PR TITLE
XDP ethernet packet

### PR DESCRIPTION
**Pull request by shankarseal**
XDP hook must indicate ethernet packets to attached programs. But current implementation had the context data buffer at the beginning of IP header. This PR fixes that as follows:
- Change the WFP layer for L2 callout to INBOUND_MAC_FRAME_NATIVE.
- Modify droppacket.c to handle ethernet frames.
- Modify unit test code to handle the above changes.

Note: this PR should be merged after #492 .

### CODE REVIEW

**Review by dthaler — COMMENTED** [review]
Looks good, will approve once the other PR is merged and this one is rebased.

**Comment by shankarseal** [comment]
I caused a bad rebase. Fixing.

**Review by Alan-Jowett — APPROVED** [review]

File: tools/encode_program_info/encode_program_info.cpp
Comment (dthaler) [inline tools/encode_program_info/encode_program_info.cpp:9]: This file needs to be reverted.    PR#492 put the correct defines into encode_program_info.h which is included at line 7.
Below has the incorrect defines with the _ prefix.

Comment (shankarseal) [reply tools/encode_program_info/encode_program_info.cpp]: fixed.

**Review by dthaler — APPROVED** [review]

**Pull request merged**
